### ISSUE TITLE
"history: true" adds multiple previous states on the Back button of the browser

### ISSUE THREAD

**Issue by pnpetroff** [andreknieriem/simplelightbox#68]
I'm using the following code

```js
function initSimpleLightbox(){

	if ($lightboxes != '') {
		$lightboxes.destroy();
	}

	$('.is-lightbox-gallery').each(function(){
		var $this = $(this);

		$lightboxes = $this.find('*:visible a');

		$lightboxes.simpleLightbox({
			showCounter: false,
			animationSpeed: 150,
			history: true,
			closeText: '<span class="close-icon"></span>',
			navText: ['<span class="arrow-left"></span>', '<span class="arrow-right"></span>']
		});
	});
}

initSimpleLightbox();
```

Every time the page loads, there are multiple previous states under the Back button of the browser. Is this a bug in the plugin or am I not writing doing right?

**Comment by andreknieriem** [comment]
Thanks, will take a look

**Comment by andreknieriem** [comment]
Hm I don't unterstand. Do you have a link or jsfiddle? Thanks!

**Comment by andreknieriem** [comment]
long no answer, closed for now

**Comment by boczpeter** [comment]
1. Suppose you have SL working on page "gallery.html" with history:true set.
2. Then, you open the gallery and push "gallery.html#pid-1" to the history stack.
3. Then, you close the gallery, pushing "gallery.html" to the stack.
Now your history stack consists of "gallery.html", "gallery.html#pid-1", "gallery.html" and your current position in the stack being the last element.
Usability-wise its a problem, because if you want to click Back button to navigate to the page preceding your first "gallery.html" in the stack, you'd expect to get there via a single click, but actually you need 3 back clicks.
Thanks for checking it.

**Comment by leoliu** [comment]
Come here to say that this bug is still in the latest release. Can we re-open this issue?

**Comment by andreknieriem** [comment]
Hey, alright, I will see what is going on.

**Comment by Alamot** [comment]
One can use history.go(-1) after some calls to close() to ameliorate this issue:
```
$('.sl-wrapper').on('click.'+prefix+ ' touchstart.'+prefix, '.sl-close', function(e) {
  e.preventDefault(); if(opened) { close(); history.go(-1); }
});

if (key == 27) { close(); history.go(-1);  }
```

**Comment by andreknieriem** [comment]
Fixed in version 2.0.0
